### PR TITLE
test/common: simplify and further explain cleanup conditions

### LINKED ISSUE #3754: Improve readability of e2e framework if statement
The complex multi-line `if-statement` needs to to be simplified so this is understood by people reading the code.

A few techniques that we could use:

- `!(A == B)` could become `A != B`
- `td.InstType == KindCluster` could be `isKindCluster`
- complex multiline statements could be evaluated and assigned to a variable, with a self-explanatory name, ahead of time

https://github.com/openservicemesh/osm/blob/93c9be585f8bbd7e295f3ff1c07233bbf41c9a34/tests/framework/common.go#L1088-L1092

### PR DESCRIPTION
**Pull request by eduser25**
There was a set of conditions a bit convoluted. It has been separated
into a function and a check with additional isolated comments to
hopefully better explain the conditions.

Fixes #3754

Tests                      [x] 

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?
No
1. Is this a breaking change?
No

### CODE REVIEW

File: tests/framework/common.go [hunk 1]
```diff
@@ -1173,6 +1170,25 @@ func (td *OsmTestData) CreateDockerRegistrySecret(ns string) {
 	}
 }
 
+// ShouldCleanupK8sOnKind returns whether k8s cleanup can be avoided on kind when the
+// kind cluster is not going to be kept anyway, when the test/suite is running on kind.
+// This optimization considerably speeds up the test suite when running on kind as we don't wait
+// for k8s cleanup after every test.
+func (td *OsmTestData) ShouldCleanupK8sOnKind(ct CleanupType) bool {
```
Comment (shashankram) [inline tests/framework/common.go:1177]: un-export this function: s/ShouldCleanupK8sOnKind/shouldCleanupK8sOnKind

File: tests/framework/common.go [hunk 2]
```diff
@@ -1173,6 +1170,25 @@ func (td *OsmTestData) CreateDockerRegistrySecret(ns string) {
...
+	cleanupK8sOnKind := false
+
+	if ct == Test {
+		// If Kind cluster is going away, no need to trigger and wait for k8s cleanup
+		cleanupK8sOnKind = !td.CleanupKindClusterBetweenTests
+	}
+	if ct == Suite {
```
Comment (shashankram) [inline tests/framework/common.go:1184]:
```suggestion
	} else if ct == Suite {
```

**Review by jaellio — APPROVED** [review]

File: tests/framework/common.go (hunk 1 above)
Comment (eduser25) [reply tests/framework/common.go:1177]: will do

File: tests/framework/common.go (hunk 2 above)
Comment (eduser25) [reply tests/framework/common.go:1184]: i doubted abut that. Being an enum I'd have preferred a switch, but we only have two of them. I guess `if...else` is actually fine.

**Review by SanyaKochhar — APPROVED** [review]

**Review by johnsonshi — APPROVED** [review]

**Pull request merged**